docs/content/docs/guide/resolver.md

```diff
@@ -174,18 +175,71 @@ If the object fetched cannot be parsed as a Tekton `Task` it will error out.
 
 ## Remote Pipeline annotations
 
-Remote Pipeline can be referenced by annotation, this allows you to share your Pipeline definition across.
+Remote Pipeline can be referenced by annotation, allowing you to share a Pipeline across multiple repositories.
 
-Only one Pipeline is allowed in annotation.
+Only one Pipeline is allowed on the `PipelineRun` annotation.
 
-An annotation to a remote pipeline looks like this :
+An annotation to a remote pipeline looks like this, using a remote URL:
 
 ```yaml
 pipelinesascode.tekton.dev/pipeline: "https://git.provider/raw/pipeline.yaml
 ```
 
-It supports remote URL and files inside the same Git repository.
+or from a relative path inside the repository:
 
-{{< hint info >}}
-[Tekton Hub](https://hub.tekton.dev) doesn't currently have support for `Pipeline`.
-{{< /hint >}}
+```yaml
+pipelinesascode.tekton.dev/pipeline: "./tasks/pipeline.yaml
+```
+
+Fetching `Pipelines` from the [Tekton Hub](https://hub.tekton.dev) is not currently supported.
+
+### Overriding tasks from a remote pipeline on a PipelineRun
+
+Remote task annotations on the remote pipeline are supported. No other
+annotations like `on-target-branch`, `on-event` or `on-cel-expression` are
+supported.
+
+If a user wants to override one of the tasks from the remote pipeline, they can do
+so by adding a task in the annotations that has the same name In their `PipelineRun` annotations.
+
+For example if the user PipelineRun contains those annotations:
+
+```yaml
+kind: PipelineRun
+metadata:
+  annotations:
+    pipelinesascode.tekton.dev/pipeline: "https://git.provider/raw/pipeline.yaml
+    pipelinesascode.tekton.dev/task: "./my-git-clone-task.yaml
+```
+
+and the Pipeline referenced by the `pipelinesascode.tekton.dev/pipeline` annotation
+in "<https://git.provider/raw/pipeline.yaml>"  contains those annotations:
+
+```yaml
+kind: Pipeline
+metadata:
+  annotations:
+    pipelinesascode.tekton.dev/task: "git-clone"
+```
+
+In this case if the `my-git-clone-task.yaml` file in the root directory is a
+task named `git-clone` it will be used instead of the `git-clone` on the remote
+pipeline that is coming from the Tekon Hub.
+
+### Tasks or Pipelines Precedence
+
+From where tasks or pipelines of the same name takes precedence?
+
+for remote Tasks, when you have a `taskRef` on a task name, pac will try to find the task in this order:
+
+1. A task matched from the PipelineRun annotations
+2. A task matched from the remote Pipeline annotations
+3. A task matched fetched from the Tekton directory
+   (the tasks from the `.tekton` directory and its subdirs are automatically included)
+
+for remote Pipelines referenced on a `pipelineRef`, pac will try to match a
+pipeline in this order:
+
+1. Pipeline from the PipelineRun annotations
+2. Pipeline from the Tekton directory (pipelines are automatically fetched from
+   the `.tekton` directory and subdirs)
```

Comment:
I have a PR like below

```
---
apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  name: remote-task-test
  annotations:
    pipelinesascode.tekton.dev/on-event: "[pull_request, push]"
    pipelinesascode.tekton.dev/on-target-branch: "[remote-annotation-test]"
    pipelinesascode.tekton.dev/max-keep-runs: "2"
    pipelinesascode.tekton.dev/pipeline: "https://github.com/savitaashture/article/blob/main/remotepipeline.yaml"
    pipelinesascode.tekton.dev/task: "https://github.com/savitaashture/article/blob/main/remotetask.yaml"
spec:
  params:
    - name: repo_url
      value: "{{ repo_url }}"
    - name: revision
      value: "{{ revision }}"
  pipelineRef:
    name: remote-pipeline
```

Remote Pipeline is like below

```
---
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: remote-pipeline
  annotations:
   pipelinesascode.tekton.dev/task: "https://github.com/savitaashture/article/blob/main/remotetask.yaml"
spec:
    tasks:
      - name: noop-task
        # taskRef: 
          # name: remote-task
        taskSpec:
          steps:
            - name: noop-task
              image: registry.access.redhat.com/ubi9/ubi-micro
              script: |
                echo "hello from main"
                exit 0
```

and remote task is

```
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: remote-task
spec:
  steps:
    - name: echo
      image: alpine
      script: |
        echo "savita"
        exit 0
```


I see output as `hello from main` but it should be `savita` right or taskspec of pipeline will have more preference ?

Reply:
it can't work on taskSpec only on taskRef

Reply:
is this by design right ?
if yes then how about mentioning this point

Reply:
Ah i see you have updated :+1: